front/src/applications/operationalStudies/hooks/useAutoUpdateProjection.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+
+import { useSelector } from 'react-redux';
+
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+
+const useAutoUpdateProjection = (
```

Comment:
Can you document the behavior?

is this supposed to do something when we load, reload, create/delete a train ?

Reply:
Done !